netket/graph/lattice.py

```diff
@@ -104,323 +111,406 @@ def get_edges(atoms_positions, cutoff, distance_atol=cutoff_tol):
     return pairs
 
 
-def create_points(basis_vectors, extent, atom_coords, pbc):
+@dataclass
+class LatticeSite:
+    """
+    Contains information about a single :class:`~netket.graph.Lattice` site.
+    """
+
+    id: int
+    """Integer ID of this site"""
+    position: PositionT
+    """Real-space position of this site"""
+    cell_coord: CoordT
+    """basis coordinates of this site"""
+    inside: bool
+    """TODO: When exactly is this needed?"""
+
+    def __repr__(self):
+        s = ", ".join(map(str, (self.id, self.cell_coord)))
+        return f"LatticeSite({s})"
+
+
+def create_sites(basis_vectors, extent, apositions, pbc):
     shell_vec = _np.zeros(extent.size, dtype=int)
     shift_vec = _np.zeros(extent.size, dtype=int)
     # note: by modifying these, the number of shells can be tuned.
     shell_vec[pbc] = 2
     shift_vec[pbc] = 1
     ranges = tuple([list(range(ex)) for ex in extent + shell_vec])
-    atoms = []
-    cellANDlabel_to_site = {}
+    sites = []
+    cell_coord_to_site = {}
     for s_cell in itertools.product(*ranges):
         s_coord_cell = _np.asarray(s_cell) - shift_vec
         if _np.any(s_coord_cell < 0) or _np.any(s_coord_cell > (extent - 1)):
```

Comment:
One more question, @VolodyaCO: Is it correct that the `inside` attribute is only needed while constructing the lattice to mark the sites in the "shell" unit cells outside the lattice extent (and thus only needs to be stored during construction because it will just be `True` for all sites remaining in the lattice after init)?